packages/expo-notifications/ios/EXNotifications/Badge/BadgeModule.swift

```diff
@@ -0,0 +1,27 @@
+//  Copyright © 2024 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+public class BadgeModule: Module {
+  public func definition() -> ModuleDefinition {
+    Name("ExpoBadgeModule")
+
+    AsyncFunction("getBadgeCountAsync") { () -> Int in
+      return EXSharedApplication().applicationIconBadgeNumber
+    }
+
+    AsyncFunction("setBadgeCountAsync") { (badgeCount: Int) -> Bool in
+      let settings = await UNUserNotificationCenter.current().notificationSettings()
+
+      if settings.badgeSetting == .enabled {
+        DispatchQueue.main.async {
+          EXSharedApplication().applicationIconBadgeNumber = badgeCount
+        }
```

Comment:
Is it necessary to run it on the main thread? In async contexts we usually don't have to worry about that. If so, I guess `getBadgeCountAsync` should do this as well as it reads that value.
Also, mixing different threading mechanisms (GCD and Swift Concurrency) is not recommended. You could do something like this:
```suggestion
        Task { @MainActor in
          EXSharedApplication().applicationIconBadgeNumber = badgeCount
        }
```

Reply:
Done